### PR TITLE
coin-utils: generate gcc11 binaries

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **coin-utils/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/coin-utils/all/conanfile.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile, tools
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.files import get, apply_conandata_patches, rm, rmdir, rename
+from conan.tools.scm import Version
+from conans import AutoToolsBuildEnvironment, tools
```
Comment (uilianries) [inline recipes/coin-utils/all/conanfile.py:6]:
```suggestion
from conans import AutoToolsBuildEnvironment
```
reimported

**Comment by uilianries** [comment]
@ericLemanissier Another internal CI job to re-generate with GCC11 has been started.

File: recipes/coin-utils/all/conanfile.py [hunk 2]
```diff
@@ -76,7 +79,7 @@ def build_requirements(self):
             self.build_requires("automake/1.16.4")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```
Comment (SpaceIm) [inline recipes/coin-utils/all/conanfile.py:82]: layout please.

File: recipes/coin-utils/all/conanfile.py [hunk 3]
```diff
@@ -76,7 +79,7 @@ def build_requirements(self):
             self.build_requires("automake/1.16.4")
```
Comment (SpaceIm) [inline recipes/coin-utils/all/conanfile.py:79]: tooL_requires instead of build_requires

File: recipes/coin-utils/all/conanfile.py [hunk 4]
```diff
@@ -100,26 +103,25 @@ def _configure_autotools(self):
         self._autotools.libs = []
         if self.settings.compiler == "Visual Studio":
```
Comment (SpaceIm) [inline recipes/coin-utils/all/conanfile.py:104]: to replace by is_msvc

File: recipes/coin-utils/all/conanfile.py [hunk 5]
```diff
@@ -62,7 +65,7 @@ def validate(self):
         # FIXME: This issue likely comes from very old autotools versions used to produce configure.
         #        It might be fixed by calling autoreconf, but https://github.com/coin-or-tools/BuildTools
         #        should be packaged and added to build requirements.
-        if hasattr(self, "settings_build") and tools.cross_building(self) and self.options.shared:
+        if hasattr(self, "settings_build") and cross_building(self) and self.options.shared:
             raise ConanInvalidConfiguration("coin-utils shared not supported yet when cross-building")
 
     def build_requirements(self):
```
Comment (SpaceIm) [inline recipes/coin-utils/all/conanfile.py:71]: build_requirements => VirtualBuildEnv almost always needed

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/coin-utils/all/conanfile.py [hunk 6]
```diff
@@ -130,17 +132,17 @@ def package(self):
...
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", f"{self.package_folder}/lib")
+        rmdir(self, f"{self.package_folder}/lib/pkgconfig")
+        rmdir(self, f"{self.package_folder}/share")
 
         if self.settings.compiler == "Visual Studio":
-            os.rename(os.path.join(self.package_folder, "lib", "libCoinUtils.a"),
-                      os.path.join(self.package_folder, "lib", "CoinUtils.lib"))
+            rename(self, f"{self.package_folder}/lib/libCoinUtils.a",
+                         f"{self.package_folder}/lib/CoinUtils.lib")
```
Comment (SpaceIm) [inline recipes/coin-utils/all/conanfile.py:141]: seems neater to use use os.path.join, why this change?

File: recipes/coin-utils/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/coin-utils/all/conanfile.py:7]: Just import tools from conans as tools_legacy to not mask tools from conan.

File: recipes/coin-utils/all/conanfile.py (hunk 6 above)
Comment (SpaceIm) [reply recipes/coin-utils/all/conanfile.py:142]: os.path.join is exactly designed to be cross-platform, why take this risk, we know os.path.join works fine, we don't know yet potential side effects of this change.

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
I can another PR to cover Conan v2 after this one. Here, we only want to re-trigger coin-utils CI job to generate missing packages.

**Comment by uilianries** [comment]
@ericLemanissier New packages has been generated with success: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/prod/TapaholesList/131/summary.json

**Comment by ericLemanissier** [comment]
@prince-chrismc @jgsogo @toge Could one of you review this PR please?

**Review by jgsogo — APPROVED** [review]
